be/src/storage/vectorized/delta_writer.cpp

```diff
@@ -124,14 +124,49 @@ Status DeltaWriter::_init() {
     }
 
     RowsetWriterContext writer_context(kDataFormatV2, config::storage_format_version);
+
+    const std::vector<SlotDescriptor*> req_slot_descriptors_without_op_col(
+            (*_req.slots).begin(), (*_req.slots).end() - ((*_req.slots).back()->col_name() == "__op"));
+
+    // maybe partial update, change to partial tablet schema
+    if (_tablet->tablet_schema().keys_type() == KeysType::PRIMARY_KEYS &&
+        req_slot_descriptors_without_op_col.size() < _tablet->tablet_schema().num_columns()) {
+        TabletSchemaPB partial_tablet_schema_pb;
+        partial_tablet_schema_pb.set_id(_tablet->tablet_schema().id());
+        partial_tablet_schema_pb.set_next_column_unique_id(_tablet->tablet_schema().next_column_unique_id());
+        partial_tablet_schema_pb.set_compress_kind(_tablet->tablet_schema().compress_kind());
+        partial_tablet_schema_pb.set_num_rows_per_row_block(_tablet->tablet_schema().num_rows_per_row_block());
+        partial_tablet_schema_pb.set_num_short_key_columns(_tablet->tablet_schema().num_short_key_columns());
+        partial_tablet_schema_pb.set_keys_type(_tablet->tablet_schema().keys_type());
+        partial_tablet_schema_pb.set_bf_fpp(_tablet->tablet_schema().bf_fpp());
+        if (_tablet->tablet_schema().has_bf_fpp()) {
+            partial_tablet_schema_pb.set_bf_fpp(_tablet->tablet_schema().bf_fpp());
+        }
+        for (const auto* slot_descriptor : req_slot_descriptors_without_op_col) {
+            const auto& slot_col_name = slot_descriptor->col_name();
+            std::size_t index = _tablet->field_index(slot_col_name);
+            if (index < 0) {
+                std::stringstream ss;
+                ss << "invalid column name: " << slot_col_name;
+                LOG(WARNING) << ss.str();
+                return Status::InternalError(ss.str());
+            }
+            _tablet->tablet_schema().column(index).to_schema_pb(partial_tablet_schema_pb.add_column());
+        }
```

Comment:
maybe create a new method for TabletSchema to create a new TabletSchema by specifying columns, and call that method instead.
std::shared_ptr<TabletSchema> TabletSchema::create(src_schema, column_names_or_ids)